src/test/testType.ts

```diff
@@ -43,7 +43,9 @@ export class TestTypeImpl {
     test.afterEach = wrapFunctionWithLocation(this._hook.bind(this, 'afterEach'));
     test.beforeAll = wrapFunctionWithLocation(this._hook.bind(this, 'beforeAll'));
     test.afterAll = wrapFunctionWithLocation(this._hook.bind(this, 'afterAll'));
-    test.skip = wrapFunctionWithLocation(this._modifier.bind(this, 'skip'));
+    const skipModifier = wrapFunctionWithLocation(this._modifier.bind(this, 'skip'));
+    const skipTest = wrapFunctionWithLocation(this._createTest.bind(this, 'skip'));
+    test.skip = (...args: any[]) => (typeof args[0] === 'string' && typeof args[1] === 'function') ? skipTest(...args as [string, Function]) : skipModifier(...args);
```

Comment:
I'd prefer a special case in `this._modifier` function. That way we'll get better diagnostics because we'll check whether we are in the collection phase or the execution phase.